### DISCUSSION TITLE
BC: Neuman condition and Robin condition

### DISCUSSION THREAD

**Discussion by ghost** [zwicker-group/py-pde#302] [Q&A]
Hello to all, 
I am trying to solve a differential diffusion equation in 1D condition where a surface reaction is taking place and in the volume, away from the diffusion layer, the concentration of the reactive species will decrease.
Because of this, I have two questions: 
1) I am looking to impose a Robin bc-left modelling this type of equation dC_dX + value * c^ 0.3 = 0 but I can't find how to modify the "classic" condition : {"type": "mixed",..... 
Do you have any idea ? 
2) I am looking to impose a Neumann bc-right condition where my temporal is a derivative dC_dt, I don't know if it is possible ? 
Thank youj in advance for your time,

G

**Comment by david-zwicker** [comment]
I'm afraid the boundary condition you're trying to impose is not yet supported, since it would require imposing a derivative, which is a rather general function of the value at the boundary itself. Such conditions can be problematic since they in principle might not be unique and it is thus difficult to implement them in such a general package.

I don't understand your second point, could you specify the boundary conditions in equations, giving all dependencies?

**Reply by david-zwicker** [reply]
I just remember that there might be a hack that gets an approximation to your first question. You can use `{"derivative_expression": "- 3 * value ** 0.3"}` to impose a derivative that depends on the value close to the boundary. Instead of using the value directly at the boundary, this uses the value at the first support point, though. This might be good enough for your use-case.

**Reply by ghost** [reply]
Thank you for this, but I don't understand where does come from the '-3' in your expression ? Could you explain to me ? 
Thanks again

**Reply by david-zwicker** [reply]
The expression "- 3 * value ** 0.3" is the quantity which is calculated for each boundary point and the outward normal derivative is then imposed to be this value. In this case `value` correspond to the value of the field, i.e., `c` in your case. I replaced the value in your expression by `3` to give a particular value for what you called `value` in your expression to avoid some confusion.

**Reply by fzmwml** [reply]
Hi

I would like to confirm that boundary conditions with normal derivative are not supported?

Is the Helmholtz equation for Dirichlet  b.c. supported.

**Reply by david-zwicker** [reply]
`py-pde` supports general boundary conditions that impose a value or a (normal) derivative at the boundary. The coupled case of a mixed condition for involving the value and the derivative is generally only support for linear conditions, i.e., standard Robin conditions.

**Comment by ghost** [comment]
Concerning my second point, in fact, one of my BC condition is kinetic rate law (with a power 0.3) and the second BC (the second point) is the diminution of my concentration in the stirred bulk. I want to impose this type of Neumann BC:
![Capture d’écran 2022-10-08 à 18 16 10](https://user-images.githubusercontent.com/115284665/194717191-26f1cb1d-2088-4a37-a618-71c4c61d0259.png)

**Reply by david-zwicker** [reply]
I don't think I have ever seen this expressions for a boundary condition and it looks a bit odd to me. However, I guess you have a PDE for ∂c/∂t, so you might be able to convert it into a proper Neumann condition.

**Comment by fzmwml** [comment]
Thank you so much for your prompt reply, David.

Babak



Get Outlook for Android<https://aka.ms/ghei36>
________________________________
